serial_asyncio_fast/__init__.py

```diff
@@ -310,7 +314,7 @@ def _write_data(self, data: Union[bytes, bytearray, memoryview]) -> None:
         except serial.SerialException as exc:
             self._fatal_error(exc, "Fatal write error on serial transport")
             return
-        except BaseException as exc:
+        except Exception as exc:
```

Comment:
I don't remember the original motivation for `BaseException` being used but this would change the exceptions caught by this method.

Is there a typing reason for this change?

Reply:
_fatal_error is expecting Exception not BaseException, I believe.

Reply:
Actually it was `_loop.call_connection_lost()`.  Sorry

Anyways, reverted here and split off into: https://github.com/home-assistant-libs/pyserial-asyncio-fast/pull/29